va/va.go

```diff
@@ -524,3 +540,30 @@ func (va VAImpl) fetchHTTP(identifier string, token string) ([]byte, string, *ac
 
 	return body, url.String(), nil
 }
+
+//this reverseaddr funntion is from net/dnsclient.go form golang
+func reverseaddr(addr string) string {
+	const hexDigit = "0123456789abcdef"
+	ip := net.ParseIP(addr)
+	if ip == nil {
+		return ""
+	}
+	// Apperently IP type in net package saves all ip in ipv6 formant, from biggest byte to smallest. we need last 4 bytes, so ip[15] to ip[12]
+	if ip.To4() != nil {
+		return strconv.Itoa(int(ip[15])) + "." + strconv.Itoa(int(ip[14])) + "." + strconv.Itoa(int(ip[13])) + "." + strconv.Itoa(int(ip[12])) + ".in-addr.arpa."
+	}
+	// Must be IPv6
+	buf := make([]byte, 0, len(ip)*4+len("ip6.arpa."))
+	// Add it, in reverse, to the buffer
+	// each part of slice has two quads, so manual hex conversion is needed and fmt %x doesn't help
```

Comment:
I'm not sure this is true, you can use `fmt.Sprintf("%x.%x", x&0f, x>>4)` which would be significantly simpler.